graphblas/matrix.py

```diff
@@ -807,13 +819,20 @@ def apply(self, op, right=None, *, left=None):
                         extra_message="Literal scalars also accepted.",
                         op=op,
                     )
-            op = get_typed_op(op, self.dtype, right.dtype, is_right_scalar=True, kind="binary")
+            if opclass in ("IndexUnaryOp", "SelectOp"):
+                op = get_typed_op(
+                    op, self.dtype, right.dtype, is_right_scalar=True, kind="indexunary"
+                )
+                cfunc_method = "IndexOp"
+            else:
+                op = get_typed_op(op, self.dtype, right.dtype, is_right_scalar=True, kind="binary")
+                cfunc_method = "BinaryOp2nd"
             if op.opclass == "Monoid":
```

Comment:
This if-else block can be indented (no need to check if opclass is an select op).  Also, we can just use `opclass` now instead of `op.opclass`.